lib/floki/finder.ex

```diff
@@ -68,18 +68,7 @@ defmodule Floki.Finder do
         case combinator do
           nil -> [html_node|acc]
           _ ->
-            case combinator.match_type do
-              :descendant ->
-                traverse(children_nodes, sibling_nodes, combinator.selector, acc)
-              :child ->
-                traverse_child(children_nodes, sibling_nodes, combinator.selector, acc)
-              :sibling ->
-                traverse_sibling(children_nodes, sibling_nodes, combinator.selector, acc)
-              :general_sibling ->
-                traverse_general_sibling(children_nodes, sibling_nodes, combinator.selector, acc)
-              other ->
-                raise "Combinator of type \"#{other}\" not implemented"
-            end
+            traverse_using(combinator.match_type, children_nodes, sibling_nodes, combinator.selector, acc)
```

Comment:
@gmile I think we can remove one argument to this function by passing the `combinator` and accessing `match_type` and `selector` from inside the function. What do you think?